src/gurt/telemetry.c

```diff
@@ -726,88 +708,83 @@ d_tm_build_path(char **path, char *item, va_list args)
 }
 
 /**
- * Compute gauge statistics, min, max, mean, and standard deviation
- * Uses B. P. Welford's algorithm for computing a running variance.
- * This implementation computes an integer min and max with floating point
- * mean and standard deviation.
+ * Compute standard deviation
+ *
+ * \param[in]	sum_of_squares	Precomputed sum of squares
+ * \param[in]	sample_size	Number of elements in the data set
+ * \param[in]	mean		Mean of all elements
+ *
+ * \return			computed standard deviation
+ */
+double
+d_tm_compute_standard_dev(double sum_of_squares, uint64_t sample_size,
+			  double mean)
+{
+	if (sample_size == 0)
+		return 0;
+
+	return sqrtl((sum_of_squares - (sample_size * mean * mean)) /
+		     (sample_size - 1));
```

Comment:
What if sample_size == 1?

Reply:
yes, I broke that with this update.  I'll fix it.